agent/agent.go

```diff
@@ -120,30 +123,55 @@ func NewAgent(ctx context.Context, appclient appclientset.Interface, namespace s
 		return nil, fmt.Errorf("unexpected agent mode: %v", a.mode)
 	}
 
-	appInformer := appinformer.NewAppInformer(ctx, appclient, a.namespace,
-		appinformer.WithListAppCallback(a.listAppCallback),
-		appinformer.WithNewAppCallback(a.addAppCreationToQueue),
-		appinformer.WithUpdateAppCallback(a.addAppUpdateToQueue),
-		appinformer.WithDeleteAppCallback(a.addAppDeletionToQueue),
-		appinformer.WithFilterChain(a.DefaultFilterChain()),
-	)
+	// appListFunc and watchFunc are anonymous functions for the informer
+	appListFunc := func(ctx context.Context, opts v1.ListOptions) (runtime.Object, error) {
+		return appclient.ArgoprojV1alpha1().Applications(a.namespace).List(ctx, opts)
```

Comment:
Should we call `Manage()` and `IgnoreChange()` on the listed apps?  I noticed that we invoke them after creating the resource in `ApplicationManager.Create()`. But I wonder if we need the `managedApps` and `observedApps` populated when the informer is initialized.

Reply:
Actually a good thought, but I think the previous iteration, i.e. having these calls in the lister, was a bug.

All items that are returned by the list function and pass the (new) informer's filter will implicitly go through the handler called for adding a resource, and the code for claiming an app to be managed is already in that handler. Those that do not pass the filter will also not be managed.

So I think the list callback is correct as is.